### PR TITLE
Ensure that the parser errors or not for the parser tests

### LINKED ISSUE #290: ra_syntax: can't parse weird-exprs.rs correctly
rust-analyzer reports lots of syntax errors on [this code](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=7cf3546d8416fc76e5828481576fe6f9) (extracted [from the compiler](https://github.com/rust-lang/rust/blob/master/src/test/run-pass/weird-exprs.rs) + added ktulhu).

### PR DESCRIPTION
**Pull request by DJMcNab**

### CODE REVIEW

**Comment by matklad** [comment]
BTW, that test is big, so it might be a good idea to send a PR per bug, not a single PR which fixes everything.

**Comment by DJMcNab** [comment]
If I do them all in one I can add `weird_exprs` to the test suite and work through the bugs one by one. I suppose I can make a PR for each if you prefer though.

**Comment by matklad** [comment]
>If I do them all in one I can add weird_exprs to the test suite and work through the bugs one by one.

You can do that, just don't commit it until it all is green, and send intermediate results

> I suppose I can make a PR for each if you prefer though.

Yeah, I slightly prefer that, but you can do a single PR if you like. It would be nice though to include individual `// test` tests for specific bugs.

**Comment by DJMcNab** [comment]
Yeah, I'll make separate PRs then.

File: crates/tools/src/main.rs [hunk 1]
```diff
@@ -48,34 +49,43 @@ fn main() -> Result<()> {
...
     let tests = tests_from_dir(Path::new(GRAMMAR_DIR))?;
+    fn install_tests(tests: &HashMap<String, Test>, into: &str, mode: Mode) -> Result<()> {
+        let tests_dir = project_root().join(into);
+        if !tests_dir.is_dir() {
+            fs::create_dir_all(&tests_dir)?;
+        }
+        // ok is never actually read, but it needs to be specified to create a Test in existing_tests
+        let existing = existing_tests(&tests_dir, true)?;
+        for t in existing.keys().filter(|&t| !tests.contains_key(t)) {
+            // panic!("Test is deleted: {}", t);
```
Comment (matklad) [inline crates/tools/src/main.rs]: Should we uncomment this?

**Comment by DJMcNab** [comment]
I fixed a couple of bugs, which got rid of error nodes. That could feasibly be the reason

**Comment by matklad** [comment]
Hm, so it's macro_stuck.txt which weighs 250 and is removed. We don't have the corresponding rs file. Looks like I've once removed the `.rs` part of the test, but left the .txt part in place.

**Comment by matklad** [comment]
bors r+

:tada:

**Comment by DJMcNab** [comment]
I've got it - https://github.com/rust-analyzer/rust-analyzer/pull/299/commits/134fe4f566d94fd4ca91c6417fab0ae7b3e4275f#diff-e146a3f1a039e47daa23fa302c22b14d. 0034_macro_stuck.txt has no .rs file.

**Pull request merged**